firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/I2C/LTC4015/Device.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020, the Pez Globo team and the Pufferfish project contributors
+ *
+ *      Author: Rohan Purohit
+ *
+ *  A driver for LTC4015 buck battery charger
+ */
+
+#include "Pufferfish/Driver/I2C/LTC4015/Device.h"
+
+#include "Pufferfish/HAL/Interfaces/Endian.h"
+#include "Pufferfish/Util/Endian.h"
+
+namespace Pufferfish::Driver::I2C::LTC4015 {
+
+I2CDeviceStatus Device::read_charging_status(bool &charging_status) {
+  std::array<uint8_t, sizeof(uint16_t)> buffer{};
+  I2CDeviceStatus ret = i2cdevice_.read(static_cast<uint16_t>(Command::system_status), buffer);
+  if (ret != I2CDeviceStatus::ok) {
+    return ret;
+  }
+
+  charging_status = false;
+  uint16_t system_status = 0;
+  Util::read_bigend(buffer.data(), system_status);
+  if ((system_status & static_cast<uint16_t>(Mask::charger_enabled)) != 0) {
+    charging_status = true;
+  }
+  return I2CDeviceStatus::ok;
+}
+
+I2CDeviceStatus Device::read_battery_voltage(uint16_t &v_bat) {
+  std::array<uint8_t, sizeof(uint16_t)> buffer{};
+  I2CDeviceStatus ret = i2cdevice_.read(static_cast<uint16_t>(Command::vbat_addr), buffer);
+  if (ret != I2CDeviceStatus::ok) {
+    return ret;
+  }
+
+  uint16_t vbat_value = 0;
+  Util::read_bigend(buffer.data(), vbat_value);
+  v_bat =
+      vbat_value *
+      static_cast<uint16_t>(
+          batt_voltage_conversion);  // https://www.analog.com/media/en/technical-documentation/data-sheets/4015fb.pdf
```

Comment:
The formatting here is weird, I think clang-format will fix it if you move the comment to its own line before the line where you assign `v_bat`. This also applies to the other comments in this file.